src/ensembl/src/content/app/custom-download/state/attributes/attributesState.ts

```diff
@@ -0,0 +1,38 @@
+import { AttributeWithOptions } from 'src/content/app/custom-download/types/Attributes';
+import JSONValue from 'src/shared/types/JSON';
+import { CheckboxGridOption } from 'src/content/app/custom-download/components/checkbox-grid/CheckboxGrid';
+import customDownloadStorageService from 'src/content/app/custom-download/services/custom-download-storage-service';
+
+export type OrthologueState = {
+  searchTerm: string;
+  species: CheckboxGridOption[];
+  showBestMatches: boolean;
+  showAll: boolean;
+  applyToAllSpecies: boolean;
+};
+
+export type AttributeContentState = {
+  [key: string]: AttributeWithOptions | { [key: string]: AttributeWithOptions };
+};
+
+export type AttributesAccordionState = Readonly<{
+  expandedPanel: string;
+  content: AttributeContentState;
+  selectedAttributes: JSONValue;
+  contentState: JSONValue;
```

Comment:
Just when I thought I was starting to understand what is going on... 😄 What is the `contentState` field in redux state and how is it different from the `content` field?

Also, why do we keep `selectedAttributes` as a separate redux field instead of updating the `isChecked` field in the options kept in the `content` field?

Reply:
Renamed `contentState` to `uiState` on ContentBuilder and to `ui` on redux.